### PR TITLE
new: enforcerlog

### PR DESCRIPTION
**Pull request by sibicramesh**
> Project https://github.com/orgs/aporeto-inc/projects/689

### CODE REVIEW

File: specs/enforcer.spec [hunk 1]
```diff
@@ -107,13 +107,15 @@ attributes:
     type: boolean
     exposed: true
     stored: true
+    deprecated: true
```
Comment (primalmotion) [inline specs/enforcer.spec]: This is not deprecated. This is how we ask enforcer to do a collection

File: specs/enforcer.spec [hunk 2]
```diff
@@ -140,6 +142,7 @@ attributes:
     type: time
     exposed: true
     stored: true
+    deprecated: true
```
Comment (primalmotion) [inline specs/enforcer.spec]: This is not deprecated. This is where enforcer reports its version

File: specs/enforcerinfo.spec [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# Attributes
+attributes:
+  v1:
+  - name: collectedInfo
+    description: Represents the latest information collected by the enforcer.
+    type: string
+    exposed: true
+    stored: true
+
+  - name: collectionTime
```
Comment (primalmotion) [inline specs/enforcerinfo.spec]: Use @timeable abstract

File: specs/enforcerinfo.spec [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  entity_name: EnforcerInfo
+  package: squall
+  group: core/enforcer
+  description: Post a new enforcer information.
+
+# Attributes
+attributes:
+  v1:
+  - name: collectedInfo
+    description: Represents the latest information collected by the enforcer.
```
Comment (primalmotion) [inline specs/enforcerinfo.spec]: Let's call that data

File: specs/enforcerinfo.spec [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Model
+model:
+  rest_name: enforcerinfo
```
Comment (primalmotion) [inline specs/enforcerinfo.spec]: enforcerlog

File: specs/enforcerinfo.spec [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    stored: true
+
+  - name: enforcerID
+    description: ID of the enforcer.
+    type: string
+    exposed: true
+    required: true
+    example_value: xxx-xxx-xxx-xxx
+
+  - name: enforcerInfoID
```
Comment (primalmotion) [inline specs/enforcerinfo.spec]: collectionID

File: specs/enforcerinfo.spec [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# Model
+model:
+  rest_name: enforcerinfo
+  resource_name: enforcerinfo
+  entity_name: EnforcerInfo
+  package: squall
+  group: core/enforcer
+  description: Post a new enforcer information.
```
Comment (primalmotion) [inline specs/enforcerinfo.spec]:
```
extends:
  - '@zoned'
  - '@migratable'
  - '@namespaced'
  - '@identifiable-stored'
  - '@timeable'
```

File: specs/enforcerinfo.spec [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
```
Comment (primalmotion) [inline specs/enforcerinfo.spec]: missing indexes on 
```
- - enforcerID
- - namespace
  - enforcerID
- - collectionID
- - namespace 
  - collectionID
```

Comment (primalmotion) [inline specs/enforcerinfo.spec]: missing internal methods

```
get:
    description: Retrieves the ...
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerlogs",
    "component": "backend",
    "pr-id": "612",
    "commit-sha": "513f9d47711b931ea7bd94f3885cdf547ea883e7"
  },
  {
    "project": "enforcerlogs",
    "component": "gaia",
    "pr-id": "403",
    "commit-sha": "f6ce05f1f7c5c960f4b0c8b84de12e2d5f34ed73"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerlogs",
    "component": "backend",
    "pr-id": "612",
    "commit-sha": "513f9d47711b931ea7bd94f3885cdf547ea883e7"
  },
  {
    "project": "enforcerlogs",
    "component": "gaia",
    "pr-id": "403",
    "commit-sha": "e634d166d3fff06561d9d04910f2b6d8b28c00ca"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerlogs",
    "component": "gaia",
    "pr-id": "403",
    "commit-sha": "e634d166d3fff06561d9d04910f2b6d8b28c00ca"
  },
  {
    "project": "enforcerlogs",
    "component": "backend",
    "pr-id": "612",
    "commit-sha": "1c8fd31792c8648ba3c1d4225d47ec941814ee6e"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerlogs",
    "component": "backend",
    "pr-id": "612",
    "commit-sha": "27f66494f2dd9bb07abb9e3a42a6b088aed89343"
  },
  {
    "project": "enforcerlogs",
    "component": "gaia",
    "pr-id": "403",
    "commit-sha": "e634d166d3fff06561d9d04910f2b6d8b28c00ca"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerlogs",
    "component": "gaia",
    "pr-id": "403",
    "commit-sha": "f3f74110caa0eb43cfaa014d74ca96bdcaa1b842"
  },
  {
    "project": "enforcerlogs",
    "component": "backend",
    "pr-id": "612",
    "commit-sha": "85f0b442fc2fdb694544b6c871fbca3309d4c4e3"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerlogs",
    "component": "backend",
    "pr-id": "612",
    "commit-sha": "85f0b442fc2fdb694544b6c871fbca3309d4c4e3"
  },
  {
    "project": "enforcerlogs",
    "component": "gaia",
    "pr-id": "403",
    "commit-sha": "f3f74110caa0eb43cfaa014d74ca96bdcaa1b842"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerlogs",
    "component": "backend",
    "pr-id": "612",
    "commit-sha": "85f0b442fc2fdb694544b6c871fbca3309d4c4e3"
  },
  {
    "project": "enforcerlogs",
    "component": "gaiajs",
    "pr-id": "26",
    "commit-sha": "dbc7ff2d7365a7ebaa4d707b60edda5747c167cf"
  },
  {
    "project": "enforcerlogs",
    "component": "clad",
    "pr-id": "1402",
    "commit-sha": "ed5ef06295d2e2a92cc2861fbf68ecc36e9de312"
  },
  {
    "project": "enforcerlogs",
    "component": "enforcerd",
    "pr-id": "1487",
    "commit-sha": "e5e07c0bf018406d66babdd2304bd953258fa797"
  },
  {
    "project": "enforcerlogs",
    "component": "gaia",
    "pr-id": "403",
    "commit-sha": "f3f74110caa0eb43cfaa014d74ca96bdcaa1b842"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerlogs",
    "component": "gaia",
    "pr-id": "403",
    "commit-sha": "f3f74110caa0eb43cfaa014d74ca96bdcaa1b842"
  },
  {
    "project": "enforcerlogs",
    "component": "backend",
    "pr-id": "612",
    "commit-sha": "85f0b442fc2fdb694544b6c871fbca3309d4c4e3"
  },
  {
    "project": "enforcerlogs",
    "component": "clad",
    "pr-id": "1402",
    "commit-sha": "ed5ef06295d2e2a92cc2861fbf68ecc36e9de312"
  },
  {
    "project": "enforcerlogs",
    "component": "enforcerd",
    "pr-id": "1487",
    "commit-sha": "e5e07c0bf018406d66babdd2304bd953258fa797"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerlogs",
    "component": "backend",
    "pr-id": "612",
    "commit-sha": "85f0b442fc2fdb694544b6c871fbca3309d4c4e3"
  },
  {
    "project": "enforcerlogs",
    "component": "enforcerd",
    "pr-id": "1487",
    "commit-sha": "e5e07c0bf018406d66babdd2304bd953258fa797"
  },
  {
    "project": "enforcerlogs",
    "component": "gaia",
    "pr-id": "403",
    "commit-sha": "f3f74110caa0eb43cfaa014d74ca96bdcaa1b842"
  }
]
```

**Pull request merged**